### PR TITLE
Support Sign Unsigned Tx

### PR DESCRIPTION
**Pull request by h0ngcha0**
UI Preview:

image

### CODE REVIEW

File: src/modals/SignUnsignedTx/index.tsx
Comment (h0ngcha0) [inline src/modals/SignUnsignedTx/index.tsx:113]: To h0ngcha0: fix translation

File: locales/en-US/translation.json
Comment (nop33) [reply locales/en-US/translation.json:329]: After adding the English entry (as you just did) you can run:

```
npm run syncTranslationFiles
```

and commit the diff.

Comment (h0ngcha0) [reply locales/en-US/translation.json:329]: I see, thanks! I see that this command populates all the language specific files with the English text. Do we need to wait until all are translated before we can merge?

Comment (mvaivre) [reply locales/en-US/translation.json:329]: No, only the English version needs to be up to date. The other translations can come later (we rely on the community for those!).

Comment (h0ngcha0) [reply locales/en-US/translation.json:329]: Thanks! :+1:

File: src/modals/SignUnsignedTx/index.tsx
Comment (mvaivre) [inline src/modals/SignUnsignedTx/index.tsx:110]: Nit: We could use the following shorter syntax:

```ts
decodedUnsignedTx && ...
```

File: src/modals/SignUnsignedTx/index.tsx
Comment (mvaivre) [inline src/modals/SignUnsignedTx/index.tsx:67]: Nit: As the `decodeUnsignedTx` function is used only once here, and for consistency sake (avoid the `.then()` syntax), we could declare and call the function right inside the `useEffect`:

```ts
useEffect(() => {
  const decodeUnsignedTx = async () => {
    setIsLoading(true)
    ...
    setDecodedUnsignedTx({
       txId: decodedResult.unsignedTx.txId,
       fromGroup: decodedResult.fromGroup,
       toGroup: decodedResult.toGroup,
       unsignedTx: txData.unsignedTx,
       gasAmount: decodedResult.unsignedTx.gasAmount,
       gasPrice: BigInt(decodedResult.unsignedTx.gasPrice)
     })
    setIsLoading(false)
 }

  decodeUnsignedTx()
, [...]}
```

File: src/modals/SignUnsignedTx/index.tsx
Comment (mvaivre) [inline src/modals/SignUnsignedTx/index.tsx:116]:
```ts
onClick={handleSign}
```

**Comment by mvaivre** [comment]
Again, great work! I've done some functional testing and it seems to work. However, I find confusing that the way the unsigned transaction is encoded seems to differ from what I see in the multi-sig tool:

image

vs 

image


As a user I was expecting to be able to compare the 2 strings.

**Comment by polarker** [comment]
@mvaivre  That's a great question. They were the same initially, but then Thomas and Polto were confused too because they used that raw tx string with Swagger.

Now I think it's good to keep them same again. Nice one!

**Comment by h0ngcha0** [comment]
> 

I guess we will update the app, right?

**Comment by h0ngcha0** [comment]
> @mvaivre That's a great question. They were the same initially, but then Thomas and Polto were confused too because they used that raw tx string with Swagger.
> 
> Now I think it's good to keep them same again. Nice one!


I guess we will update the app, right?

**Comment by h0ngcha0** [comment]
> > @mvaivre That's a great question. They were the same initially, but then Thomas and Polto were confused too because they used that raw tx string with Swagger.
> > Now I think it's good to keep them same again. Nice one!
> 
> I guess we will update the app, right?

Sorry, just saw the update on app.

**Review by polarker — CHANGES_REQUESTED** [review]
LGTM

**Review by polarker — APPROVED** [review]
LGTM

**Review by mvaivre — APPROVED** [review]

**Pull request merged**
